aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -210,7 +211,7 @@ class TORCH_API OperatorEntry final {
 
   OperatorName name_;
   c10::optional<AnnotatedSchema> schema_;
-
+  c10::optional<std::unordered_set<Tags>> tags_;
   std::array<KernelFunction, c10::num_runtime_entries> dispatchTable_;
```

Comment:
So today, `schema_` is a `c10::optional` but it's considered an error for `schema_` to be none at runtime (and we have a `DanglingImpl` test to catch that here: https://github.com/pytorch/pytorch/blob/5762c7b25b06620f95e251f2bf5d4a1564f99d03/aten/src/ATen/core/dispatch/Dispatcher.cpp#L324.

We might want to maintain the same invariant for `tags_`, so anyone using it at runtime doesn't have to worry about checking for whether the optional is invalid

Reply:
Actually, this suggests we may want to refactor the "def" info into a single struct, which has a single optional surrounding it. Otherwise, it's possible to accidentally have schema_ nullopt but tags not (or vice versa), which is illegal

Reply:
tags is no longer optional so I guess we don't need to do anything here